### PR TITLE
Pickpocketing improvements and additions

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
The imprinter might or might not come depending if I get sprites.

Contents:
- Added **Pickpocket gloves with storage**, they can hold up to two small items and pickpocketing with these gloves will automatically make the stolen item go into the gloves. Available for 7 TC, or 4 if you're an Assistant
- Original pickpocket gloves have been made cheaper, 4 TC by default and 2 if you're an Assistant. Good if you want to field other people with stealing implements.
- **Pickpocketing gloves of any kind will make you more quiet**, that means you won't make any sound to anyone when you shuffle your items around. Now you can steal something and put it in your backpack without as much as a noise or indication it ever happened.

Code review appreciated because I feel it's too much spaghetti.
It's all been tested.

:cl:
 * rscadd: Added new pickpocket gloves that come with their own storage to place stolen items in automatically.
 * tweak: Regular pickpocket gloves have been made cheaper.
 * tweak: Being a pickpocket of any kind will make your object shuffling inaudible.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
demonstration

**Comment by Irradiation** [comment]
I'm a bit dubious about the "tested" part when it comes to having nearby people not hear backpack ruffling.

**Comment by boy2mantwicethefam** [comment]
It's a bunch of oldcode but what it does is that M is actually just the user and M just shows a visible message for everyone nearby based on conditions
As for the audio itself that just means the audio won't be played even if you're within 2-3 tiles of people

**Comment by Goldsteel** [comment]
Now powergamers have even more of a reason to constantly be 1 tile away from you at all times.

**Comment by ghost** [comment]
at last

File: code/modules/clothing/gloves/color.dm [hunk 1]
```diff
@@ -102,6 +102,61 @@
...
+	var/list/can_only_hold = new/list()
+	var/list/cant_hold = list("/obj/item/clothing/gloves/black/thief/storage") //ISHYGDDT
+	var/fits_max_w_class = W_CLASS_SMALL
+	var/max_combined_w_class = 4
+	var/storage_slots = 2 //Two gloves
+
+//Copypasted storage suit code
+/obj/item/clothing/gloves/black/thief/storage/New()
+	..()
+	hold = new (src)
```
Comment (jknpj) [inline code/modules/clothing/gloves/color.dm:120]: >setting all this on new() instead of making a storage/subtype

**Pull request merged**